contacts.py

```diff
@@ -42,6 +65,7 @@ def __init__(self, directory):
         self.item_count = 0
         self.update_index = {}
         # [ (floating_seconds, serial_number)* ] used to order data by time
+        # TODO-DAN is this correct, shouldnt it sort on both floating_seconds and serial_number (though I'm not sure it matters if we consider them the same time)
```

Comment:
I'm commenting here, we can discuss tomorrow prior to merge.  No, the sort is just on the time, we don't care about the serial number because this sortedlist is used for the since <= time < until calc, no serial number needed.